File: PRG351_Java/Java_Mod_5/src/java_mod_5/Housing.java

package java_mod_5;

public class Housing implements iUnits {
        protected int m_yearBuilt;
        protected String m_Address;

        public Housing()
        {
            m_yearBuilt = 1980;
            m_Address = "234 Main St";
        }
        
        public Housing( int year, String address)
        {
            m_yearBuilt = year;
            m_Address = address;
        }

        @Override 
        public int NumberUnits()
        {
            return 4;
        }
        
        //@Override To String is never defined in any parent class
        // Therefore it is non-sesical to attempt at redefining the 
        // behavior of ToString()
        public String ToString()
        {
            String s = "Year built = " + m_yearBuilt + " - Address = " + m_Address;
            return s;
        }
        
        public double GetRentalAmount()
        {
            return 1565.5;
        }
}

File: PRG321/week02/p717N/Housing.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace p717N10
{
    public class Housing : IUnits
    {
        protected int m_yearBuilt;
        protected string m_Address;

        public Housing()
        {
            m_yearBuilt = 1980;
            m_Address = "234 Main St";
        }

        virtual public int NumberUnits()
        {
            return 4;
        }

        public override string ToString()
        {
            String s = "Year built = " + m_yearBuilt + " - Address = " + m_Address;
            return s;
        }

        public virtual double GetRentalAmount()
        {
            return 1565.5;
        }
    }
}
